var/spack/repos/builtin/packages/openfoam-com/package.py

```diff
@@ -657,36 +658,35 @@ def __init__(self, spec, **kwargs):
         if '+float32' in spec:
             self.precision_option = 'SP'
 
-        # spec.architecture.platform is like `uname -s`, but lower-case
-        platform = spec.architecture.platform
+        # OpenFOAM chooses which wmake rules to use based on the output of
+        # `uname -s` and `uname -m`.  We can get the same information from
+        # Python's platform library.
 
-        # spec.architecture.target is like `uname -m`
-        target   = spec.architecture.target
+        system = platform.system().lower()
+        machine = platform.machine()
 
-        if platform == 'linux':
-            if target == 'i686':
+        self.arch = system
+        self.arch_option = '64'
+
+        if system == 'linux':
+            if machine == 'i686':
                 self.arch_option = '32'  # Force consistency
-            elif target == 'x86_64':
-                if self.arch_option == '64':
-                    platform += '64'
-            elif target == 'ia64':
-                platform += 'ia64'
-            elif target == 'armv7l':
-                platform += 'ARM7'
-            elif target == 'ppc64':
-                platform += 'PPC64'
-            elif target == 'ppc64le':
-                platform += 'PPC64le'
-        elif platform == 'darwin':
-            if target == 'x86_64':
-                platform += 'Intel'
+            elif machine == 'x86_64':
+                self.arch += '64'
```

Comment:
To change between 32 and 64 arch_option would actually require a hand-edit of the package file. I don't suggest that anyone wants this, but it does reflect the backend OpenFOAM configuration possibility. If there a need did happen to arise for 32-bit on 64-bit arch, this could be controlled as a spack variant. At the moment I don't think we need this extra variant (more clutter, probably not used), but left this in a config constant for documentation purposes and future-proofing.